Ladybird/Android/src/main/cpp/LadybirdActivity.cpp

```diff
@@ -13,27 +14,26 @@
 #include <jni.h>
 
 OwnPtr<Core::EventLoop> s_main_event_loop;
-RefPtr<Core::Timer> s_timer;
 
 extern "C" JNIEXPORT void JNICALL
-Java_org_serenityos_ladybird_LadybirdActivity_initNativeCode(JNIEnv* env, jobject /* thiz */, jstring resource_dir, jobject timer_service)
+Java_org_serenityos_ladybird_LadybirdActivity_initNativeCode(JNIEnv* env, jobject /* thiz */, jstring resource_dir, jstring tag_name, jobject timer_service)
 {
     char const* raw_resource_dir = env->GetStringUTFChars(resource_dir, nullptr);
     s_serenity_resource_root = raw_resource_dir;
     __android_log_print(ANDROID_LOG_INFO, "Ladybird", "Serenity resource dir is %s", s_serenity_resource_root.characters());
     env->ReleaseStringUTFChars(resource_dir, raw_resource_dir);
+    dbgln("Set resource dir to {}", s_serenity_resource_root);
+
+    char const* raw_tag_name = env->GetStringUTFChars(tag_name, nullptr);
+    AK::set_log_tag_name(MUST(String::from_utf8(StringView { raw_tag_name, strlen(raw_tag_name) })));
```

Comment:
Isn't it kind of pointless to drop the `\0` that's already there here, only to append it again in `set_log_tag_name`? On the other hand I guess you have to copy the string anyway?

Reply:
I suppose, but we don't have that `CString` class in AK yet, so this is the best I got :). Without passing a StringView that I pinky promise will always be called with a null term string, which seems fragile

Reply:
I mean, just pass a `char const*`?

Reply:
:facepalm: I've been looking at C++ too long